src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -928,6 +928,8 @@ private static void EnsureInitialized()
         /// <summary>Ensures that the console has been initialized for use.</summary>
         private static void EnsureInitializedCore()
         {
+            Debug.Assert(!Monitor.IsEntered(Console.InternalSyncObject)); // see LazyInitializeConsoleAndFieldCore
```

Comment:
Left-over comment?

Reply:
I'll update the comment, or remove the check depending on @stephentoub's feedback.